org.idempiere.test/src/org/idempiere/test/base/MTableTest.java

```diff
@@ -396,6 +474,557 @@ public void testMarkImmutable() {
 		} catch (Exception e) {}
 		assertTrue(cached.isActive(), "IsActive after setIsActive should be active, no change");
 		assertSame(tbl, cached, "Cached table must remain consistent after markImmutable");
+		
+		MColumn[] columns = tbl.getColumns(false);
+		MColumn[] cachedColumns = cached.getColumns(false);
+		if (cachedColumns.length > 0) {
+			try {
+				cachedColumns[0].setIsActive(false);
+			} catch (Exception e) {}
+			assertTrue(cachedColumns[0].isActive(), "IsActive after setIsActive should be active, no change");
+			assertSame(columns, cachedColumns, "Cached columns must remain consistent after markImmutable");
+		}
+	}
+
+	/**
+	 * Test cases for MTable.get(int AD_Table_ID).
+	 */
+	@Test
+	public void testGet_ByTableID() {
+	    try {
+	        // Valid table ID - use an existing core table such as AD_Table
+	        MTable existingTable = MTable.get(Env.getCtx(), MTable.Table_Name, getTrxName());
+	        assertNotNull(existingTable, "AD_Table must exist in the system");
+	        int validTableId = existingTable.getAD_Table_ID();
+
+	        MTable result1 = MTable.get(validTableId);
+	        assertNotNull(result1, "Valid AD_Table_ID should return a table");
+	        assertEquals(MTable.Table_Name, result1.getTableName(), "Returned table must be AD_Table");
+
+	        // Invalid table ID
+	        int invalidTableId = 9999999;
+	        MTable result2 = MTable.get(invalidTableId);
+	        assertNull(result2, "Invalid table ID should return null");
+
+	        // Zero table ID
+	        MTable result3 = MTable.get(0);
+	        assertNotNull(result3, "AD_Table_ID=0 should return a table");
+	        assertNull(result3.getTableName(), "Should return null for AD_Table_ID=0");
+	    } finally {
+	        rollback();
+	    }
+	}
+	
+	/**
+	 * Test cases for MTable.getClass(String tableName).
+	 */
+	@Test
+	public void testGetClass() {
+	    try {
+	        Class<?> clazz1 = MTable.getClass(MWebService.Table_Name);
+	        assertEquals(MWebService.class, clazz1, "Should return class from cached factory");
+
+	        Class<?> clazz2 = MTable.getClass(MWebService.Table_Name);
+	        assertEquals(MWebService.class, clazz2, "Should return class from second valid factory");
+
+	        assertThrows(NullPointerException.class,
+		            () -> MTable.getClass("DoesNotExist"),
+		            "should throw NullPointerException");
```

Comment:
@etantg , this test is wrong, it should return null, not throw exception. This is actually a bug in the DefaultModelFactory class, line 87, it should just return null if table is null.